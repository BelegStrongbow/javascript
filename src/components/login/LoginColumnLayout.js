import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const ColumnLayout = styled.div`
	
`;

const ColumnLeft = styled.div`
	float: left;
	width: 480px;
`;

const ColumnRight = styled.div`
	/* 480 + 48px left padding */
	margin-left: 528px;
`;

/**
 * A one or two column layout for a login / registration / reset password page.
 *
 * First child component is displayed in the left column, second is displayed on the right.
 *
 * Reverts to a one column layout when only one child is given.
 * When more then two child components are given, only the first two are shown.
 *
 * @returns {ReactElement} A two column layout containing the two children as columns.
 */
class LoginColumnLayout extends React.Component {

	constructor() {
		super();
	}

	getRightColumn() {
		return <ColumnRight>
			{ this.props.children[ 1 ] }
		</ColumnRight>;
	}

	render() {
		let twoColumns = this.props.children.length > 1;

		return (
			<ColumnLayout>
				<ColumnLeft>
					{ this.props.children[ 0 ] }
				</ColumnLeft>
				{ twoColumns ? this.getRightColumn() : null }
			</ColumnLayout>
		);
	}
}

LoginColumnLayout.propTypes = {
	children: PropTypes.any,
};

export default LoginColumnLayout;

