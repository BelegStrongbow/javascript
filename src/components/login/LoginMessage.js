import React from "react";
import PropTypes from "prop-types";
import { injectIntl, FormattedMessage } from "react-intl";
import styled from "styled-components";

// Components.
import { Button } from "../Button";

// Images.
import icon from "../../images/greeting.png";
import logo from "../../images/my-yoast-academy-logo.svg";

const MainPaper = styled.div`
	padding: 0px;
	width: 480px;
	height: 480px;

    text-align: center;
`;

const Icon = styled.img`
	margin-top: 20px;
	width: 200px;
`;

const Logos = styled.img`
	margin-top: 40px;
	width: 360px;
`;

const Message = styled.p`
	padding: 0 2em;
`;

/**
 * A card representing a message in a login screen,
 * e.g. "Welcome back!", "Password changed".
 *
 * Contains the MyYoast and Yoast Academy logos,
 * an image (defaults to a standard 'greeting' image) and a header.
 *
 * If an onClick handler is given, a button is rendered with the given message,
 * if none is given, the message is rendered as a paragraph.
 *
 * @returns {ReactElement} The component that contains the message
 */
class LoginMessage extends React.Component {

	constructor( props ) {
		super( props );
		this.renderButton = this.renderButton.bind( this );
		this.renderParagraph = this.renderParagraph.bind( this );
	}

	/**
	 * Creates a button with the message as given in the props.
	 * Adds the onClick callback (also given in the props).
	 *
	 * @returns {React.Component} the button, with the onClick callback added.
	 */
	renderButton() {
		return (
			<Button onClick={ this.props.onClick }>
				<FormattedMessage { ...this.props.message } />
			</Button>
		);
	}

	/**
	 * Renders a paragraph with the message given in the props.
	 *
	 * @returns {React.Component} the paragraph, with the message as its contents.
	 */
	renderParagraph() {
		return (
			<Message>
				<FormattedMessage { ...this.props.message } />
			</Message>
		);
	}

	render() {
		return (
			<MainPaper>
				<Logos src={ logo } alt="MyYoast - Yoast Academy" />
				<Icon src={ this.props.image } alt="" />
				<h2>
					<FormattedMessage id={ this.props.header.id }
									  defaultMessage={ this.props.header.defaultMessage } />
				</h2>
				{ this.props.onClick ? this.renderButton() : this.renderParagraph() }
			</MainPaper>
		);
	}
}

LoginMessage.propTypes = {
	image: PropTypes.string,
	header: PropTypes.object,
	message: PropTypes.object,
	onClick: PropTypes.func,
};

LoginMessage.defaultProps = {
	image: icon,
};

export default injectIntl( LoginMessage );
