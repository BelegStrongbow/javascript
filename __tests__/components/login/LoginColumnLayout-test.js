import React from 'react';
import LoginMessage from "../../../src/components/login/LoginMessage";
import { defineMessages } from "react-intl";
import { createComponentWithIntl } from "../../../utils";
import LoginColumnLayout from "../../../src/components/login/LoginColumnLayout";

const messages = defineMessages( {
	header: {
		id: "login.header",
		defaultMessage: "Welcome back!",
	},
	message: {
		id: "login.message",
		defaultMessage: "Log in with your email address and password. If you don't remember your password, just reset it!",
	},
	headerReset: {
		id: "login.headerReset",
		defaultMessage: "Password changed successfully!",
	},
	button: {
		id: "login.button",
		defaultMessage: "Continue to MyYoast",
	},
} );

test('The LoginColumnLayout component, with two columns, matches the snapshot', () => {
	const component = createComponentWithIntl(
		<LoginColumnLayout>
			<LoginMessage header={ messages.header } message={ messages.message }/>
			<LoginMessage header={ messages.headerReset } message={ messages.button } buttonLinkTo="https://example.com"/>
		</LoginColumnLayout>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('The LoginColumnLayout component, with one column, matches the snapshot', () => {
	const component = createComponentWithIntl(
		<LoginColumnLayout>
			<LoginMessage header={ messages.header } message={ messages.message }/>
		</LoginColumnLayout>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
