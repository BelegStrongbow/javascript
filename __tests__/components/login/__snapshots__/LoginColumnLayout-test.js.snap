// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LoginColumnLayout component, with one column, matches the snapshot 1`] = `
.c2 {
  padding: 0px;
  width: 480px;
  height: 480px;
  text-align: center;
}

.c4 {
  margin-top: 20px;
  width: 200px;
}

.c3 {
  margin-top: 40px;
  width: 360px;
}

.c5 {
  padding: 0 2em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 480px;
}

@media screen and ( max-width:1024px ) {
  .c1:not(:only-child) {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        className="c3"
        src="my-yoast-academy-logo.svg"
      />
      <img
        className="c4"
        src="greeting.png"
      />
      <h2>
        <span>
          Welcome back!
        </span>
      </h2>
      <p
        className="c5"
      >
        <span>
          Log in with your email address and password. If you don't remember your password, just reset it!
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`The LoginColumnLayout component, with two columns, matches the snapshot 1`] = `
.c8 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  background-color: #64a60a;
  color: #fff;
  box-shadow: inset 0 -4px 0 rgba(0,0,0,0.2);
  -webkit-filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  border-radius: 4px;
  font: 700 14px/24px "Open Sans",sans-serif;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c8:hover,
.c8:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c8:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c7 {
  text-shadow: 0px 0px 2px #000;
}

.c2 {
  padding: 0px;
  width: 480px;
  height: 480px;
  text-align: center;
}

.c4 {
  margin-top: 20px;
  width: 200px;
}

.c3 {
  margin-top: 40px;
  width: 360px;
}

.c5 {
  padding: 0 2em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 480px;
}

.c6 {
  margin-left: 48px;
}

@media screen and ( max-width:1024px ) {
  .c1:not(:only-child) {
    display: none;
  }
}

@media screen and ( max-width:1024px ) {
  .c6 {
    margin-left: 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        className="c3"
        src="my-yoast-academy-logo.svg"
      />
      <img
        className="c4"
        src="greeting.png"
      />
      <h2>
        <span>
          Welcome back!
        </span>
      </h2>
      <p
        className="c5"
      >
        <span>
          Log in with your email address and password. If you don't remember your password, just reset it!
        </span>
      </p>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="c2"
    >
      <img
        className="c3"
        src="my-yoast-academy-logo.svg"
      />
      <img
        className="c4"
        src="greeting.png"
      />
      <h2>
        <span>
          Password changed successfully!
        </span>
      </h2>
      <button
        className="c7 c8"
        onClick={[Function]}
        type="button"
      >
        <span>
          Continue to MyYoast
        </span>
      </button>
    </div>
  </div>
</div>
`;
